src/erlcloud_ddb_streams.erl

```diff
@@ -0,0 +1,907 @@
+    {error, Reason};
+request_to_return(#aws_request{response_type = error,
+                               error_type = httpc,
+                               httpc_error_reason = Reason}) ->
+    {error, Reason}.
+
+-spec result_fun(#aws_request{}) -> #aws_request{}.
+result_fun(#aws_request{response_type = ok} = Request) ->
+    Request;
+result_fun(#aws_request{response_type = error,
+                        error_type = aws,
+                        response_status = Status,
+                        response_status_line = StatusLine,
+                        response_body = Body} = Request) ->
+    %% == IMPLEMENTATION NOTES ==
+    %% We store the error reason in `httpc_error_reason` for now,
+    %% this may be changed at any time later
+    Request2 = Request#aws_request{
+        httpc_error_reason = {http_error, Status, StatusLine, Body}},
+    if
+        Status >= 400 andalso Status < 500 ->
+            client_error(Request2);
+        Status >= 500 ->
+            Request2#aws_request{should_retry = true};
+        Status < 400 ->
+            Request2#aws_request{should_retry = false}
+    end.
+
+-spec client_error(#aws_request{}) -> #aws_request{}.
+client_error(#aws_request{response_body = Body} = Request) ->
+    case jsx:is_json(Body) of
+        false ->
+            Request#aws_request{should_retry = false};
+        true ->
+            Json = jsx:decode(Body),
+            case proplists:get_value(<<"__type">>, Json) of
+                undefined ->
+                    Request#aws_request{should_retry = false};
+                FullType ->
+                    Message = proplists:get_value(<<"message">>, Json, <<>>),
+                    case binary:split(FullType, <<"#">>) of
+                        [_, Type] when
+                              Type =:= <<"LimitExceededException">>;
+                              Type =:= <<"ThrottlingException">> ->
+                            Request#aws_request{httpc_error_reason = {Type, Message},
+                                                should_retry = true};
+                        [_, Type] ->
+                            Request#aws_request{httpc_error_reason = {Type, Message},
+                                                should_retry = false};
+                        _ ->
+                            Request#aws_request{should_retry = false}
+                    end
+            end
+    end.
+
+-spec headers(aws_config(), string(), binary()) -> [{string(), string()}].
+headers(Config, Operation, Body) ->
+    Headers = [{"host", Config#aws_config.ddb_streams_host},
+               {"x-amz-target", Operation},
+               {"content-type", "application/x-amz-json-1.0"}],
+    Region =
+        case string:tokens(Config#aws_config.ddb_streams_host, ".") of
+            [_, _, Value, _, _] ->
+                Value;
+            _ ->
+                "us-east-1"
+        end,
+    erlcloud_aws:sign_v4(Config, Headers, Body, Region, "dynamodbstreams").
```

Comment:
Oops, I misread the aws java sdk source code. It turned out the service name came from [`awssdk_config_default.json`](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/resources/awssdk_config_default.json#L185), not [inferred from the class name](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceClient.java#L681-L709). Sorry for this.